wordpress-mobile/WordPress-Android
# Refactor PostListButton type to enum

Converts PostListButton to kotlin and migrates PostListButtonType to an enum.

To test:
- since we are merging this into a working branch and we'll test all the logic anyway I think it's not necessary to do any through-out tests. 
1. My Site
2. Open Blog Posts
3. Make sure the action buttons are displayed
4. Click on one of the buttons and make sure it works as expected

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
